Overhaul of verify mode and added typeCheck mode

MatthewDaggitt opened the pull request:
See CHANGELOG entries for description

MatthewDaggitt committed:
Overhaul of verify mode and added typeCheck mode

MatthewDaggitt committed:
Updated hashes

wenkokke commented:
I'd ask to call the command `check` for brevity?

Isn't typeCheck mode already exposed via `compile -t TypeCheck`?

MatthewDaggitt committed:
Fixes 'ignore' field bugs in TestSpec and ignores file hashes in golden files

MatthewDaggitt committed:
Rename 'check' mode to 'validate' and 'typeCheck' to 'check'

MatthewDaggitt commented:
> I'd ask to call the command check for brevity?

The mode `check` was already reserved for checking the proof-cache, but in order to follow your suggestion I have renamed that command `validate`, and changed `typeCheck` to `check`.

> Isn't typeCheck mode already exposed via compile -t TypeCheck?

Yes, but semantically this makes far more sense as now the mode `compile` and its target option `-t` really is reserved for compilation targets.

MatthewDaggitt committed:
Merge branch 'dev' into mode-overhaul
